### PR TITLE
Create RepositoryRefsOptions and ListRefs method with helper funds

### PR DESCRIPTION
**Pull request by DataDavD**
TL;DR: this PR creates a RepositoryRefsOptions type and starts the groundwork to eventually remove RepositoryBranchOptions and RepositoryTagOptions and those methods that relies on these types. 

This PR creates a RepositoryRefsOptions and a ListRefs method on the Repository type. This new refs type will allow users to specific their refs options once and be able to operate on all kinds of refs (i.e. tags and branches) without specifying 2 separate options. Right now, users have to specify a RepositoryBranchOptions and RepositoryTagOptions in order to call ListBranches and ListTags. 

This current iteration of the code doesn't adhere to the fact that git refers to all branches and tags as refs and this is reflected in the Bitbucket API since both tags and branches endpoints are build off of the refs endpoint. 

I want to add a test for this new ListRefs function, but in the meantime I wanted to create this PR s that we could discuss future plans. Currently, I think it is best to keep Refs options alongside the branches and tags options since if we remove those 2 option types then it imparts a breaking change on users. So we should slowly move towards removing those option types and their methods instead removing support all at once and making a bunch of breaking changes.

Once we have built out the same functionality for branches and tags build upon the refs options type then we can move to completely remove the branches and tags options types and release for a breaking change.

### CODE REVIEW

**Comment by ktrysmt** [comment]
Exactly. First, we should add the new apis in it. after that, we update old apis to show deprecated notifications for an end user. Finally, delete the old apis.
So, I think that only this PR should create new api first. @DataDavD

**Comment by ktrysmt** [comment]
And, Looks good for me, may be.

**Comment by ktrysmt** [comment]
I'll scrutinize it later.

**Comment by DataDavD** [comment]
Hey @ktrysmt FYI I added a test case for the ListRefs method. I only ran the `test_repository.go` test cases which passed. My test isn't the greatest but it works. I believe the PR is good now. I will work on adding the delete branch and tag methods next in https://github.com/ktrysmt/go-bitbucket/issues/154.

**Review by ktrysmt — APPROVED** [review]
LGTM! Thank you.

**Comment by DataDavD** [comment]
Great thank you @ktrysmt. I'll look out for the merge! 💯 . I'm going to start working on the delete branch and tag methods which I think will be useful for setting up testing that tests methods that work on branches/tags.
